v0/destinations/am/transform.js

```diff
@@ -161,21 +164,43 @@ function updateConfigProperty(
             }
           }
         } else {
-          // get the destKey/outKey value from calling the util function
-          set(payload, outKey, AMUtils[funcName](message, sourceKey, Config));
+          const data = get(message.traits, outKey); // when in identify(or any other call) it checks whether outKey is present in traits
+          // then that value is assigned else function is applied.
+          // that key (outKey) will be a default key for wh actions and thus removed from the payload.
+          if (isDefinedAndNotNull(data)) {
+            set(payload, outKey, data);
+            delete message.traits[outKey];
+          } else {
+            // get the destKey/outKey value from calling the util function
+            set(payload, outKey, AMUtils[funcName](message, sourceKey, Config));
+          }
         }
       }
     } else {
+      // For common config
       if (validatePayload) {
-        const data = get(payload, mappingJson[sourceKey]);
-        if (!isDefinedAndNotNull(data)) {
-          const val = get(message, sourceKey);
-          if (val || val === false || val === 0) {
-            set(payload, mappingJson[sourceKey], val);
+        // if data is present in traits assign
+        const messageData = get(message.traits, mappingJson[sourceKey]);
+        if (isDefinedAndNotNull(messageData)) {
+          set(payload, mappingJson[sourceKey], messageData);
+        } else {
+          const data = get(payload, mappingJson[sourceKey]);
+          if (!isDefinedAndNotNull(data)) {
+            const val = get(message, sourceKey);
+            if (val || val === false || val === 0) {
+              set(payload, mappingJson[sourceKey], val);
+            }
+          } else {
```

Comment:
Do we need this `else` condition? Since data is already present from line 187.

Reply:
in line 187 data value is assigned.
If the value is undefined or null it goes it the if. 
so if the value is present we are adding this else to handle that

Reply:
I meant, the `payload` has the value of `data` with key `mappingJson[sourceKey]` in line 187.
In `else` condition we are just setting the same value which is already present in payload.

Reply:
missed it yes. Will remove this.

Reply:
done